core/network/corda-interop-app/interop-workflows/src/main/kotlin/com/weaver/corda/app/interop/flows/AssetExchangeHTLCFlows.kt

```diff
@@ -651,5 +652,10 @@ fun resolveUpdateOwnerFlow(flowName: String, flowArgs: List<Any>): Either<Error,
 
 fun getLinearIdFromString(linearId: String): UniqueIdentifier {
     val id = linearId.split("_").toTypedArray()
+    if (id.size != 2) {
+        // here the id[1] denotes an UUID and id[0] denotes its hash
+        println("Invalid linearId: ${linearId}.\n")
+        throw IllegalStateException("Invalid linearId: ${linearId}.\n")
```

Comment:
Can you modify message to be like "Invalid LinearId as per HTLC or Asset Transfer Corda Protocol"? So that in future nobody gets confused again, that they have a perfect linearId but this function isn't working.
It's just in HTLC and Asset Transfer we are appending Hash to the linearId.

Reply:
Done.